sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/inventory/SimInventoryResource.kt

```diff
@@ -111,15 +112,21 @@ class SimInventoryResource(private val api: SimInventoryApi) {
     fun importBatch(
             @NotEmpty @PathParam("hssVendors") hss: String,
             @NotEmpty @PathParam("simVendor") simVendor: String,
-            csvInputStream: InputStream): Response =
-            api.importBatch(hss, simVendor, csvInputStream)
-                    .fold(
-                            {
-                                error("Failed to upload batch with SIM profiles for HSS ${hss} and SIM profile vendor ${simVendor}",
-                                        ApiErrorCode.FAILED_TO_IMPORT_BATCH, it)
-                            },
-                            { Response.status(Response.Status.OK).entity(asJson(it)) }
-                    ).build()
+            @QueryParam("initialHssState") initialHssStateArg: HssState?,
+            csvInputStream: InputStream): Response {
+
+        val initialHssState = if (initialHssStateArg==null) HssState.NOT_ACTIVATED else initialHssStateArg
```

Comment:
Since this is at input I guess that it could be an idea to check whether the `initialHssStateArg` value, if set, is a valid value? That is matches one of the entries in the `HssState` enum.

Reply:
It's implicitly checked by the input filter for the applet.   If it's set it will automatically be translated to a  legal value otherwise a HTTP 500 will be thrown (will add explicit test for that).  I don't know if it's possible to test for illegal query parameters to be set or if they fail silently.  I will check.  It would be better to fail than to fail silently, because silent failure of this kind will actually import the data but with the wrong value for initial hss state set.  Good catch.

Reply:
fixed now.